### PR TITLE
[SYCL] Report exceptions during wait for dependencies of host-tasks

### PR DESCRIPTION
**Pull request by s-kanaev**
The issue was located by eyeballing

### CODE REVIEW

File: sycl/source/detail/scheduler/commands.cpp [hunk 1]
```diff
@@ -185,14 +185,27 @@ class DispatchHostTask {
...
+      try {
+        PluginWithEvents.first->call<PiApiKind::piEventsWait>(RawEvents.size(),
+                                                              RawEvents.data());
+      } catch (const sycl::exception &E) {
+        CGHostTask &HostTask = static_cast<CGHostTask &>(MThisCmd->getCG());
+        HostTask.MQueue->reportAsyncException(std::current_exception());
+        return (pi_result)E.get_cl_code();
+      } catch (...) {
+        CGHostTask &HostTask = static_cast<CGHostTask &>(MThisCmd->getCG());
+        HostTask.MQueue->reportAsyncException(std::current_exception());
```
Comment (romanovvlad) [inline sycl/source/detail/scheduler/commands.cpp:197]: Can we just call `call_nocheck` and then construct a proper exception if result is not `SUCCESS`?
Also can we add exception handling higher in the stack, so we can catch all the error and put them to the async queues?

File: sycl/source/detail/scheduler/commands.cpp [hunk 2]
```diff
@@ -201,12 +214,22 @@ class DispatchHostTask {
...
     CGHostTask &HostTask = static_cast<CGHostTask &>(MThisCmd->getCG());
 
+    pi_result WaitResult = waitForEvents();
+    if (WaitResult != PI_SUCCESS) {
+      std::exception_ptr EPtr = std::make_exception_ptr(sycl::runtime_error(
+          std::string("Couldn't wait for host-task's dependencies"),
+          WaitResult));
+      HostTask.MQueue->reportAsyncException(EPtr);
+
+      // reset host-task's lambda and quit
```
Comment (romanovvlad) [inline sycl/source/detail/scheduler/commands.cpp:228]: It looks like we will report async exception two times, one in `waitForEvents`, the second is here.

Comment (s-kanaev) [reply sycl/source/detail/scheduler/commands.cpp]: Fixed.

File: sycl/source/detail/scheduler/commands.cpp (hunk 1 above)
Comment (s-kanaev) [reply sycl/source/detail/scheduler/commands.cpp]: Fixed.

**Review by romanovvlad — APPROVED** [review]

**Comment by bader** [comment]
Test?

**Comment by bader** [comment]
Also there is a regression on Windows platform. Please, take a look.

**Comment by bader** [comment]
@s-kanaev, ping.

**Comment by bader** [comment]
@s-kanaev, are you still planning to make this change or we can close this PR?

**Pull request closed without merging**